pkg/services/session/storage/persistent/storage.go

```diff
@@ -54,7 +62,38 @@ func NewTokenStore(path string, opts ...Option) (*TokenStore, error) {
 		return nil, fmt.Errorf("could not init session bucket: %w", err)
 	}
 
-	return &TokenStore{db: db, l: cfg.l}, nil
+	ts := &TokenStore{db: db, l: cfg.l}
+
+	// enable encryption if it
+	// was configured so
+	if cfg.privateKey != nil {
+		rawKey, err := x509.MarshalECPrivateKey(cfg.privateKey)
+		if err != nil {
+			return nil, fmt.Errorf("could not marshal provided private key: %w", err)
+		}
+
+		// tagOffset is a constant offset for
+		// tags when marshalling ECDSA key in
+		// ASN.1 DER form
+		const tagOffset = 7
+
+		// using first 32 bytes from
+		// the marshalled private key
+		// as a secret
+		c, err := aes.NewCipher(rawKey[tagOffset : tagOffset+32])
```

Comment:
https://github.com/golang/go/blob/master/src/crypto/x509/sec1.go#L58
```
rawKey := make([]byte, (key.Curve.Params().N.BitLen()+7)/8)
cfg.privateKey.D.FillBytes(rawKey)
```

Also works for every private key type, not just 32-byte one.

Reply:
looks better, does the same, thanks